demo/3000-home/pages/index.js

```diff
@@ -1,12 +1,158 @@
+import React from 'react';
+import Head from 'next/head';
 import dynamic from 'next/dynamic';
-const page = import('./index.real');
-
-const Page = dynamic(() => import('./index.real'));
-Page.getInitialProps = async (ctx) => {
-  const getInitialProps = (await page).default?.getInitialProps;
-  if (getInitialProps) {
-    return getInitialProps(ctx);
-  }
-  return {};
+
+const CheckoutTitle = dynamic(() => import('checkout/CheckoutTitle'), {
```

Comment:
test if this works with non async import like 

if(process.browser) require(checkout/checkoutTitle) 

Need to ensure that it works without async boundaries

Reply:
No, it doesn't.

Fails with the following error
<img width="1041" alt="Screen Shot 2022-09-15 at 12 51 09" src="https://user-images.githubusercontent.com/1946920/190334870-6d0ed2cb-0d68-4927-934e-ff432ab52529.png">
<img width="1734" alt="Screen Shot 2022-09-15 at 12 50 03" src="https://user-images.githubusercontent.com/1946920/190334909-914a8df4-d674-4a25-9640-1d63d1adb940.png">

Definitely, MFClient does not control this area. It ensures only if the page module remoteEntry is loaded. But in this case, we are opening a home application (host), and MFClient does nothing about awaiting checkout remote here. MFClient works lazily with remotes and loads only what is needed and does not await host pages.

Reply:
In other words, here is opened a HOST page and MFClient do not add async boundaries.

If MFClient will substitute the HOST page with its REMOTE analog (in home app use `home-fed` module) then it might be possible. But I think it can be moved to another issue.

Reply:
Got it. So we still need to figure out automatic async boundary. No problem just checking